pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -778,9 +794,12 @@ protected void handleConnect(CommandConnect connect) {
             // Not find provider named authMethod. Most used for tests.
             // In AuthenticationDisabled, it will set authMethod "none".
             if (authenticationProvider == null) {
-                authRole = getBrokerService().getAuthenticationService().getAnonymousUserRole()
```

Comment:
What is the value of this authRole if the `AnonymousUserRole` is not set? is it null. We should avoid any lists where null is an element.

```
final String anonymousUserRole = getBrokerService().getAuthenticationService().getAnonymousUserRole().orElseThrow(() -> new AuthenticationException("No anonymous role, and no authentication provider " + "configured")));

authRoles = StringUtils.isNotEmpty(anonymousUserRole) ? Collections.singletonList(anonymousUserRole) : Collections.emptyList()

```